pulp_container/app/token_verification.py

```diff
@@ -167,8 +163,32 @@ def has_permission(self, request, view):
         """
         Decide upon permission based on token
         """
-        try:
-            decoded_token = request.auth
-            return _contains_accessible_actions(decoded_token, *_access_scope(request))
-        except Exception:
+        decoded_token = request.auth
+        if decoded_token is None:
             return False
+
+        try:
+            scope = init_scope(request)
+        except HasNoScope:
+            return is_requesting_root_endpoint(decoded_token)
+        else:
+            return contains_accessible_actions(decoded_token, scope)
+
+
+def is_requesting_root_endpoint(decoded_token):
+    """
+    Returns True if no access type was detected.
+    """
+    return len(decoded_token["access"]) == 0
+
+
+def contains_accessible_actions(decoded_token, scope):
+    """
+    Check if a client has permission to perform operations within the current scope
+    """
+    for access in decoded_token["access"]:
+        if scope.resource_type == access["type"] and scope.name == access["name"]:
+            if scope.action in access["actions"]:
+                return True
+
+    return False
```

Comment:
I think, i have seen this function before. Maybe you just moved it around, maybe you changed it on the way. Can you move it back, so it's easier to review?